File: 242.Valid_Anagram/Valid_Anagram_Solution.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        
        container_s, container_t = {}, {}
        for i in range(len(s)):
            container_s[s[i]]  = 1 + container_s.get(s[i],0)
            container_t[t[i]]  = 1 + container_t.get(t[i],0)
        for j in container_s:
            if container_s[j] != container_t.get(j,0):
                return False
        return True
            

File: 242.Valid_Anagram/Valid_Anagram_Solution.cpp
#include <bits/stdc++.h>
#include <vector>
#include<iostream>
class Solution {
public:
    bool isAnagram(std::string s, std::string t) {
        if(s.size() != t.size()){
            return false;
        }

        std::unordered_map<char,int> s_container;
        std::unordered_map<char,int> t_container;

        for(int i=0; i < s.size(); i++){
            s_container[s[i]]++;
            t_container[t[i]]++;
        }

        for(int j = 0; j < s_container.size(); j++){
            if(s_container[j] != t_container[j]){
                return false;
            }
        }
        return true;
    }
};

int main(int argc, char* argv[]) {
    Solution sol;
    std::string a = "anagram";
    std::string b = "nagaram";
    std::cout<<sol.isAnagram(a,b)<<std::endl;
    return 0;
}